lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -98,3 +98,13 @@ function gutenberg_register_wp_rest_templates_controller_plugin_field() {
 	);
 }
 add_action( 'rest_api_init', 'gutenberg_register_wp_rest_templates_controller_plugin_field' );
+
+/**
+ * Overrides the default 'WP_REST_Server' class.
+ *
+ * @return string The name of the custom server class.
+ */
+function gutenberg_override_default_rest_server() {
+	return 'Gutenberg_REST_Server';
+}
+add_filter( 'wp_rest_server_class', 'gutenberg_override_default_rest_server' );
```

Comment:
Won't this break any pre-existing code that declares their own custom `WP_REST_Server`?

Example usage in existing plugin directory: https://wpdirectory.net/search/01J5R1B8MEEW01RJ3WZDC2XPY8

Reply:
The same is true for every REST API override in the Gutenberg plugin. Come with the project that builds a new WP in a plugin.

We can use a low-priority filter here (`0` or `1`) and prioritize other integrations, but this would mean not shipping enhancement for those plugins.

As far as I know, server overrides should be rare in plugins unless they're shipping their custom server.

Reply:
A low priority sounds like a good compromise. Hopefully, there will be little to no overlap.

Reply:
I updated the hook priority. I also checked most of the plugins from the directory results, and none of them actually override the server class. But there will be some cases that aren't part of the plugin directory.

Reply:
Better be safe than sorry. Nice to see that most of these plugin directory instances were actually tests.